lib/logstash/filters/de_dot.rb

```diff
@@ -57,9 +77,17 @@ def rename_field(event, fieldref)
   def filter(event)
     @separator = '][' if @nested
     @logger.debug? && @logger.debug("de_dot: Replace dots with separator", :separator => @separator)
-    @fields = event.to_hash.keys if @fields.nil?
-    @logger.debug? && @logger.debug("de_dot: Act on these fields", :fields => @fields)
-    @fields.each { |ref| rename_field(event, ref) if !(ref =~ /\./).nil? }
+    if @fields.nil?
+      fields = event.to_hash.keys
+    else
+      fields = @fields
+    end
+    @logger.debug? && @logger.debug("de_dot: Act on these fields", :fields => fields)
+    fields.each do |ref|
+      if event[ref]
+        rename_field(event, ref) if !(ref =~ /\./).nil?
```

Comment:
I see we are using  `if !(ref =~ /\./).nil?` in two different places we could probably wrap that into a predicate. What do you think?

```ruby
dotted_key?(ref) or dotted?(ref), has_dot?(ref)
```